packages/x-date-pickers/src/AdapterMoment/index.ts

```diff
@@ -90,4 +94,12 @@ export class AdapterMoment
   public getWeekNumber = (date: defaultMoment.Moment) => {
     return date.week();
   };
+
+  public getWeekdays = () => {
+    return defaultMoment.weekdaysShort(true);
```

Comment:
It seems to work well with locale because moment has a global localization management 👍

([tested](https://codesandbox.io/s/competent-joji-p3b9ky?file=/demo.tsx) with fr which starts weeks on Monday instead of Sunday for the USA)

Reply:
This method is just copy pasted from `@date-io` but using `defaultMoment` instead of `this.moment`
This should not have any impact on the locale, the behavior should be the same before and after this PR.

But yes, this method is problematic with locale, and is why we have a warning [on the doc](https://next.mui.com/x/react-date-pickers/adapters-locale/#with-moment) saying to people that they should set the global locale to always equal the one passed to the `LocalizationProvider`.
And maybe it would be worth changing the locale just before this function and the re-setting it to its previous value.
But that's an independent topic I think :+1: